### ISSUE TITLE
How to install this project?

### ISSUE THREAD

**Issue by jhzao82** [mcekovic/tennis-crystal-ball#273]
I use comand gradlew.bat, It seem success, but I can't find jar file in build path. I don't How to install and run this project. Can you give a guide?

**Comment by mcekovic** [comment]
There is some info at https://github.com/mcekovic/tennis-crystal-ball/issues/232

**Comment by jhzao82** [comment]
Thank you @mcekovic ! It's a little confuse to me, But I think I can try to get start!

**Comment by jhzao82** [comment]
Hello, @mcekovic，I nearly install full, But I got this error below:
Deprecated Gradle features were used in this build, making it incompatible with Gradle 5.0.
See https://docs.gradle.org/4.7/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 49s
16 actionable tasks: 16 executed
F:\tcb\tennis-crystal-ball>cd F:\tcb\tennis-crystal-ball\data-load\build\distributions\data-load-1.0-SNAPSHOT\bin

F:\tcb\tennis-crystal-ball\data-load\build\distributions\data-load-1.0-SNAPSHOT\bin>data-load.bat -lt -Dtcb.db.connections=15 -Dtcb.data.base-dir=
Loading Tennis Data
Allocating DB connections.
Loading players
Loading file '/tcb/tennis_atp\atp_players.csv'
Exception in thread "main" java.lang.IllegalArgumentException
        at java.util.concurrent.ThreadPoolExecutor.<init>(ThreadPoolExecutor.java:1307)
        at java.util.concurrent.ThreadPoolExecutor.<init>(ThreadPoolExecutor.java:1195)
        at java.util.concurrent.Executors.newFixedThreadPool(Executors.java:89)
        at java_util_concurrent_Executors$newFixedThreadPool.call(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:128)
        at org.strangeforest.tcb.dataload.BaseCSVLoader$_load_closure1.doCall(BaseCSVLoader.groovy:66)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:98)
        at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
        at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:264)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1034)
        at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:41)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:128)
        at org.strangeforest.tcb.dataload.SqlPool.withTx(SqlPool.groovy:51)
        at org.strangeforest.tcb.dataload.SqlPool$withTx$0.callStatic(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallStatic(CallSiteArray.java:55)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callStatic(AbstractCallSite.java:197)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callStatic(AbstractCallSite.java:217)
        at org.strangeforest.tcb.dataload.SqlPool.withSql(SqlPool.groovy:42)
        at org.strangeforest.tcb.dataload.SqlPool$withSql.call(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:128)
        at org.strangeforest.tcb.dataload.BaseCSVLoader.load(BaseCSVLoader.groovy:65)
        at org.strangeforest.tcb.dataload.BaseCSVLoader$load$0.callCurrent(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:51)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:157)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:169)
        at org.strangeforest.tcb.dataload.BaseCSVLoader.loadFile(BaseCSVLoader.groovy:39)
        at org.strangeforest.tcb.dataload.BaseCSVLoader$loadFile.call(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:128)
        at org.strangeforest.tcb.dataload.ATPTennisLoader.loadPlayers(ATPTennisLoader.groovy:22)
        at org.strangeforest.tcb.dataload.ATPTennisLoader$loadPlayers.call(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:128)
        at org.strangeforest.tcb.dataload.LoadATPTennis.run(LoadATPTennis.groovy:11)
        at org.strangeforest.tcb.dataload.LoadATPTennis$run.call(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:120)
        at org.strangeforest.tcb.dataload.Main.run(Main.groovy:36)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:98)
        at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1225)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1034)
        at org.codehaus.groovy.runtime.InvokerHelper.invokePogoMethod(InvokerHelper.java:947)
        at org.codehaus.groovy.runtime.InvokerHelper.invokeMethod(InvokerHelper.java:930)
        at org.codehaus.groovy.runtime.InvokerHelper.runScript(InvokerHelper.java:428)
        at org.codehaus.groovy.runtime.InvokerHelper$runScript.call(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:136)
        at org.strangeforest.tcb.dataload.Main.main(Main.groovy)

**Comment by mcekovic** [comment]
You need to specify the location of the Jeff's tennis data repository (that you cloned from GitHub) in JVM system property -Dtcb.data.base-dir=C:\LocationOfTennisDataRepo  (seems you left it blank?)

**Comment by mcekovic** [comment]
Now seems directory is set, but seems you edited built data-load.bat file. Instead, it is probably better to keep it as it is, and specify JVM parameter overrides in the command line itself.

**Comment by jhzao82** [comment]
I edit build.gradle in data-load directory:

applicationDefaultJvmArgs = [
	'-Xmx256m',
	'-Dtcb.db.url=jdbc:postgresql://localhost:5432/tcb?prepareThreshold=0',
	'-Dtcb.db.username=tcb',
	'-Dtcb.db.password=tcb',
	'-Dtcb.db.connections=1',
	'-Dtcb.data.base-dir=F:\\tcb\\tennis_atp'
]

I also try to add this parameter to command line, It's still have problem. So I try:

F:\tcb\tennis-crystal-ball\data-load\build\distributions\data-load-1.0-SNAPSHOT\bin>data-load.bat -lt -Dtcb.db.connections=15 -Dtcb.data.base-dir=F:\\tcb\\tennis_atp
Loading Tennis Data
Allocating DB connections.
Loading players
Loading file 'F:\tcb\tennis_atp\atp_players.csv'
Exception in thread "main" java.lang.IllegalArgumentException
        at java.util.concurrent.ThreadPoolExecutor.<init>(ThreadPoolExecutor.java:1307)
        at java.util.concurrent.ThreadPoolExecutor.<init>(ThreadPoolExecutor.java:1195)
        at java.util.concurrent.Executors.newFixedThreadPool(Executors.java:89)
        at java_util_concurrent_Executors$newFixedThreadPool.call(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:128)
        at org.strangeforest.tcb.dataload.BaseCSVLoader$_load_closure1.doCall(BaseCSVLoader.groovy:66)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:98)
        at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
        at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:264)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1034)
        at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:41)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:128)
        at org.strangeforest.tcb.dataload.SqlPool.withTx(SqlPool.groovy:51)
        at org.strangeforest.tcb.dataload.SqlPool$withTx$0.callStatic(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallStatic(CallSiteArray.java:55)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callStatic(AbstractCallSite.java:197)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callStatic(AbstractCallSite.java:217)
        at org.strangeforest.tcb.dataload.SqlPool.withSql(SqlPool.groovy:42)
        at org.strangeforest.tcb.dataload.SqlPool$withSql.call(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:128)
        at org.strangeforest.tcb.dataload.BaseCSVLoader.load(BaseCSVLoader.groovy:65)
        at org.strangeforest.tcb.dataload.BaseCSVLoader$load$0.callCurrent(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:51)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:157)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:169)
        at org.strangeforest.tcb.dataload.BaseCSVLoader.loadFile(BaseCSVLoader.groovy:39)
        at org.strangeforest.tcb.dataload.BaseCSVLoader$loadFile.call(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:128)
        at org.strangeforest.tcb.dataload.ATPTennisLoader.loadPlayers(ATPTennisLoader.groovy:22)
        at org.strangeforest.tcb.dataload.ATPTennisLoader$loadPlayers.call(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:128)
        at org.strangeforest.tcb.dataload.LoadATPTennis.run(LoadATPTennis.groovy:11)
        at org.strangeforest.tcb.dataload.LoadATPTennis$run.call(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:120)
        at org.strangeforest.tcb.dataload.Main.run(Main.groovy:36)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:98)
        at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1225)
        at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1034)
        at org.codehaus.groovy.runtime.InvokerHelper.invokePogoMethod(InvokerHelper.java:947)
        at org.codehaus.groovy.runtime.InvokerHelper.invokeMethod(InvokerHelper.java:930)
        at org.codehaus.groovy.runtime.InvokerHelper.runScript(InvokerHelper.java:428)
        at org.codehaus.groovy.runtime.InvokerHelper$runScript.call(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:136)
        at org.strangeforest.tcb.dataload.Main.main(Main.groovy)

What dose this error mean?

**Comment by jhzao82** [comment]
@mcekovic any soulstion?

**Comment by jhzao82** [comment]
@mcekovic , I have a solustion in linux, database was installed, what could I do next? I want to run up the web server, any more info?

**Comment by jhzao82** [comment]
@mcekovic , I get run this, I can see this web in my host, why I can't see players? rankings table also are empty?

**Comment by jhzao82** [comment]
It's Ok now, Thank you!

**Comment by darkFunction** [comment]
I'm having the exact same issue on MacOS, could you explain how you fixed it? Thanks

**Comment by mcekovic** [comment]
Please pull the latest changes and please see the updated instructions at #232 (basically system properties are replaced with command line arguments)

**Comment by darkFunction** [comment]
Actually there were a couple of issues installing which I have now managed to resolve

Firstly the `dataload` script doesn't seem to accept arguments properly, I didn't investigate why but just hardcoded them into the defaults for my purposes.

Secondly I had to modify `BaseCSVLoader.groovy` and rebuild as the date format doesn't appear to be correct. I'm confused how this ever worked, it's very strange- here is my change:

```
-                       new Date(new SimpleDateFormat('yyyy-MM-dd').parse(d).time)
+                       new Date(new SimpleDateFormat('yyyyMMdd').parse(d).time)
```

Thirdly there were separate data cleaning issues with the Jeff Sackman repo's `csv` files, a stray string in one file and a stray character in another. Easily fixed though.
